fix: make cookies more secure

Thunderbottom opened the pull request:
* add HTTPOnly to `sid`, so `sid` will no longer be accessible through javascript.
* mark all cookies as "Secure" by default over HTTPS.
* set "SameSite" to "Strict" for all cookies by default, preventing cross-origin cookie access.
* remove redundant `publish_realtime` for setting `csrf_token` on the client-side.

Thunderbottom committed:
fix: make cookies more secure

* add HTTPOnly to sid, so sid will no longer be accessible through
  javascript.
* mark all cookies Secure by default over HTTPS.
* set SameSite to Strict for all cookies by default, preventing
  cross-origin cookie access.
* remove redundant publish_realtime for setting csrf_token on the
  client-side.

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

revant reviewed (commented):
Tried locally. Works as expected.

![cookie-stonks](https://user-images.githubusercontent.com/1040161/85032667-b7366a80-b19d-11ea-88c6-4298010e9241.png)

browser console:

```js
document.cookie;
"system_user=yes; user_image=; full_name=Administrator; user_id=Administrator"
```

Need to test on https server

Thunderbottom committed:
chore: improve logic for checking logged in user

frappe.session.user sometimes gets set after the function runs, so
current implementation would return undefined !== "Guest" as True,
causing Guest user to be set as a logged-in user.

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

adityahase reviewed (approved):
Secure flag works too.

![image](https://user-images.githubusercontent.com/8528887/87269374-a0f4a380-c4ea-11ea-97af-09408221978e.png)

**Note**: Reverse Proxy needs to set `X-Forwarded-Proto` for this to work properly. 


Bench nginx confiuration does this by default with
```
proxy_set_header X-Forwarded-Proto $scheme;
```
So no issue there.
